bindings/c/CMakeLists.txt

```diff
@@ -283,107 +283,96 @@ if(NOT WIN32)
       )
-      set_tests_properties("${test_name}" PROPERTIES TIMEOUT 300)
     endforeach()
 
-    add_test(NAME fdb_c_upgrade_to_future_version
+    add_scripted_fdb_test(NAME fdb_c_upgrade_to_future_version
       COMMAND ${CMAKE_SOURCE_DIR}/tests/TestRunner/upgrade_test.py
       --build-dir ${CMAKE_BINARY_DIR}
       --test-file ${CMAKE_SOURCE_DIR}/bindings/c/test/apitester/tests/upgrade/MixedApiWorkloadMultiThr.toml
-      --upgrade-path "7.3.0" "7.4.0" "7.3.0"
+      --upgrade-path "${FDB_CURRENT_VERSION}" "${FDB_FUTURE_VERSION}" "${FDB_CURRENT_VERSION}"
       --process-number 3
       )
-    set_tests_properties("fdb_c_upgrade_to_future_version" PROPERTIES ENVIRONMENT "${SANITIZER_OPTIONS}")
 
-    add_test(NAME fdb_c_upgrade_to_future_version_blob_granules
+    add_scripted_fdb_test(NAME fdb_c_upgrade_to_future_version_blob_granules
       COMMAND ${CMAKE_SOURCE_DIR}/tests/TestRunner/upgrade_test.py
       --build-dir ${CMAKE_BINARY_DIR}
       --test-file ${CMAKE_SOURCE_DIR}/bindings/c/test/apitester/tests/upgrade/ApiBlobGranulesCorrectness.toml
-      --upgrade-path "7.3.0" "7.4.0" "7.3.0"
+      --upgrade-path "${FDB_CURRENT_VERSION}" "${FDB_FUTURE_VERSION}" "${FDB_CURRENT_VERSION}"
       --blob-granules-enabled
       --process-number 3
       )
 
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND NOT USE_SANITIZER)
-      add_test(NAME fdb_c_client_config_tests
+      add_scripted_fdb_test(NAME fdb_c_client_config_tests
         COMMAND $<TARGET_FILE:Python3::Interpreter> ${CMAKE_CURRENT_SOURCE_DIR}/test/fdb_c_client_config_tests.py
         --build-dir ${CMAKE_BINARY_DIR}
         --client-config-tester-bin $<TARGET_FILE:fdb_c_client_config_tester>
         )
 
-      add_test(NAME fdb_c_upgrade_single_threaded_630api
+      add_scripted_fdb_test(NAME fdb_c_upgrade_single_threaded_prev3_api
         COMMAND ${CMAKE_SOURCE_DIR}/tests/TestRunner/upgrade_test.py
         --build-dir ${CMAKE_BINARY_DIR}
         --test-file ${CMAKE_SOURCE_DIR}/bindings/c/test/apitester/tests/upgrade/MixedApiWorkloadSingleThr.toml
-        --upgrade-path "6.3.23" "7.0.0" "7.1.9" "7.3.0"
+        --upgrade-path "${FDB_PREV3_RELEASE_VERSION}" "${FDB_PREV2_RELEASE_VERSION}" "${FDB_PREV_RELEASE_VERSION}" "${FDB_CURRENT_VERSION}"
         --process-number 1
         )
 
-      add_test(NAME fdb_c_upgrade_single_threaded_700api
+      add_scripted_fdb_test(NAME fdb_c_upgrade_single_threaded_prev2_api
         COMMAND ${CMAKE_SOURCE_DIR}/tests/TestRunner/upgrade_test.py
         --build-dir ${CMAKE_BINARY_DIR}
         --test-file ${CMAKE_SOURCE_DIR}/bindings/c/test/apitester/tests/upgrade/MixedApiWorkloadSingleThr.toml
-        --upgrade-path "7.0.0" "7.1.9" "7.3.0"
+        --upgrade-path "${FDB_PREV2_RELEASE_VERSION}" "${FDB_PREV_RELEASE_VERSION}" "${FDB_CURRENT_VERSION}"
         --process-number 1
         )
 
-      add_test(NAME fdb_c_upgrade_multi_threaded_630api
+      add_scripted_fdb_test(NAME fdb_c_upgrade_multi_threaded_prev3_api
```

Comment:
is it intentional that multithreaded vs singlethreaded tests differ not only in `--process_number` but also in that the multithreaded tests also do downgrades (despite the name)?

Reply:
The single-threaded tests were originally added to test upgrade back from the versions that did not support multithreading (6.2 and older). At the end, the 6.2 was removed from the upgrade path because of its flakiness and because did not want to fix a version that is so old. Not sure why we are not testing downgrades here. Possibly there is not much additional value of maintaining these single-threaded tests anymore.